### ISSUE TITLE
v2023.0.0-M6 - Issue with buildWar / buildJar

### ISSUE THREAD

**Issue by codeconsole** [graceframework/grace-framework#319]
### Describe the bug

git clone https://github.com/codeconsole/grace-website

```
./gradle bootJar
```
or
```
./gradle bootWar
```

results in the command hanging and stuck running `assetCompile`.

### Expected behavior

a jar or war is built

### Actual behavior

```
...

024-06-07T07:00:10.470-0700 [INFO] [asset.pipeline.gradle.GradleEventListener] Processing File 31 of 40 - bootstrap.js.map
2024-06-07T07:00:10.610-0700 [INFO] [asset.pipeline.gradle.GradleEventListener] Processing File 32 of 40 - bootstrap.min.js.map
2024-06-07T07:00:10.654-0700 [INFO] [asset.pipeline.gradle.GradleEventListener] Processing File 33 of 40 - bootstrap-icons.woff2
2024-06-07T07:00:08.780-0700 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
2024-06-07T07:00:08.780-0700 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :assetCompile
2024-06-07T07:00:08.780-0700 [DEBUG] [org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter] Putting task artifact state for task ':assetCompile' into context took 0.0 secs.
2024-06-07T07:00:08.780-0700 [DEBUG] [org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep] Implementation for task ':assetCompile': asset.pipeline.gradle.AssetCompile_Decorated@a5636d73e298c5eadedc7edf6223871d
2024-06-07T07:00:08.780-0700 [DEBUG] [org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep] Additional implementations for task ':assetCompile': [asset.pipeline.gradle.AssetCompile_Decorated@a5636d73e298c5eadedc7edf6223871d]
2024-06-07T07:00:08.781-0700 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Resolve files of :runtimeClasspath' completed
2024-06-07T07:00:08.781-0700 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Snapshot inputs and outputs before executing task ':assetCompile'' completed
2024-06-07T07:00:08.781-0700 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Resolve files of :runtimeClasspath' completed
2024-06-07T07:00:08.781-0700 [INFO] [org.gradle.internal.execution.steps.ResolveCachingStateStep] Caching disabled for task ':assetCompile' because:
  Build cache is disabled
2024-06-07T07:00:08.782-0700 [DEBUG] [org.gradle.internal.execution.steps.SkipUpToDateStep] Determining if task ':assetCompile' is up-to-date
2024-06-07T07:00:08.782-0700 [INFO] [org.gradle.internal.execution.steps.SkipUpToDateStep] Task ':assetCompile' is not up-to-date because:
  Output property 'destinationDir' file /Users/sheiberg/Documents/workspace/grace-framework/website/build/assets has been removed.
  Output property 'destinationDir' file /Users/sheiberg/Documents/workspace/grace-framework/website/build/assets/apple-touch-icon-9fa9b768c81f8044a2b09812c4a82db5.png has been removed.
  Output property 'destinationDir' file /Users/sheiberg/Documents/workspace/grace-framework/website/build/assets/apple-touch-icon-retina-04a78405897f12e5117a6721c820960e.png has been removed.
2024-06-07T07:00:08.782-0700 [DEBUG] [org.gradle.internal.vfs.impl.AbstractVirtualFileSystem] Invalidating VFS paths: [/Users/sheiberg/Documents/workspace/grace-framework/website/build/assets]
2024-06-07T07:00:08.782-0700 [DEBUG] [org.gradle.internal.execution.steps.CreateOutputsStep] Ensuring directory exists for property destinationDir at /Users/sheiberg/Documents/workspace/grace-framework/website/build/assets
2024-06-07T07:00:08.782-0700 [DEBUG] [org.gradle.api.internal.tasks.execution.TaskExecution] Executing actions for task ':assetCompile'.
2024-06-07T07:00:08.782-0700 [DEBUG] [asset.pipeline.fs.FileSystemAssetResolver] Asset Pipeline FSResolver Initialized with Scan Directories: [/Users/sheiberg/Documents/workspace/grace-framework/website/app/assets/images]
2024-06-07T07:00:08.782-0700 [DEBUG] [asset.pipeline.fs.FileSystemAssetResolver] Asset Pipeline FSResolver Initialized with Scan Directories: [/Users/sheiberg/Documents/workspace/grace-framework/website/app/assets/javascripts]
2024-06-07T07:00:08.782-0700 [DEBUG] [asset.pipeline.fs.FileSystemAssetResolver] Asset Pipeline FSResolver Initialized with Scan Directories: [/Users/sheiberg/Documents/workspace/grace-framework/website/app/assets/fonts]
2024-06-07T07:00:08.783-0700 [DEBUG] [asset.pipeline.fs.FileSystemAssetResolver] Asset Pipeline FSResolver Initialized with Scan Directories: [/Users/sheiberg/Documents/workspace/grace-framework/website/app/assets/stylesheets]
2024-06-07T07:00:08.783-0700 [DEBUG] [asset.pipeline.fs.FileSystemAssetResolver] Asset Pipeline FSResolver Initialized with Scan Directories: []
2024-06-07T07:00:08.783-0700 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Resolve files of :runtimeClasspath' completed
2024-06-07T07:00:10.930-0700 [LIFECYCLE] [asset.pipeline.gradle.GradleEventListener] 
2024-06-07T07:00:10.930-0700 [INFO] [asset.pipeline.gradle.GradleEventListener] Processing File 34 of 40 - bootstrap-icons.woff
2024-06-07T07:00:11.822-0700 [INFO] [asset.pipeline.gradle.GradleEventListener] Processing File 35 of 40 - bootstrap-reboot.min.css.map
2024-06-07T07:00:12.055-0700 [INFO] [asset.pipeline.gradle.GradleEventListener] Processing File 36 of 40 - bootstrap-grid.css.map
2024-06-07T07:00:12.331-0700 [INFO] [asset.pipeline.gradle.GradleEventListener] Processing File 37 of 40 - bootstrap.css.map
2024-06-07T07:00:12.572-0700 [INFO] [asset.pipeline.gradle.GradleEventListener] Processing File 38 of 40 - bootstrap.min.css.map
2024-06-07T07:00:14.193-0700 [INFO] [asset.pipeline.gradle.GradleEventListener] Processing File 39 of 40 - bootstrap-grid.min.css.map
2024-06-07T07:00:17.526-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2024-06-07T07:00:17.526-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2024-06-07T07:00:17.527-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2024-06-07T07:00:17.527-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2024-06-07T07:00:17.527-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2024-06-07T07:00:17.527-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2024-06-07T07:00:27.526-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2024-06-07T07:00:27.527-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2024-06-07T07:00:27.527-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2024-06-07T07:00:27.527-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
2024-06-07T07:00:27.527-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2024-06-07T07:00:27.527-0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.

...

2024-06-07T07:52:21.623-0700 [DEBUG] [org.gradle.launcher.daemon.client.DaemonClientConnection] thread 1: dispatching class org.gradle.launcher.daemon.protocol.Finished
2024-06-07T07:52:21.623-0700 [DEBUG] [org.gradle.launcher.daemon.client.DaemonClientConnection] thread 1: connection stop
2024-06-07T07:52:21.635-0700 [LIFECYCLE] [org.gradle.launcher.cli.DebugLoggerWarningAction] 
#############################################################################
   WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING

   Debug level logging will leak security sensitive information!

   https://docs.gradle.org/7.6.3/userguide/logging.html#sec:debug_security
#############################################################################
```

### How to Reproduce?

1. `./gradlew bootJar`

### Grace version (ie. output of `grace -v`)

```
../bin/grace --version

------------------------------------------------------------
Grace 2023.0.0-M6
------------------------------------------------------------

Build time:   2024-05-31 17:17:19 UTC
Revision:     f53b3c7b12696f308f9773ee7d6f14d8a9ce4cad

Spring Boot:  3.0.7
Gradle:       7.6.1
Groovy:       4.0.12
JVM:          17.0.10 (Oracle Corporation 17.0.10+11-LTS-jvmci-23.0-b27)
OS:           Mac OS X 14.3.1 aarch64


```

**Comment by rainboyan** [comment]
@codeconsole Thanks for your report.

This issue was fixed via https://github.com/grace-plugins/grace-asset-pipeline/commit/520e6ad302b30bd1b033a7bcbcae1e71b7cdce58

Feel free to give it an early try!

```gradle
repositories {
    mavenCentral()
    maven { url "https://s01.oss.sonatype.org/content/repositories/snapshots/" }
}

dependencies {
    implementation "org.graceframework.plugins:asset-pipeline-gradle:6.0.0-SNAPSHOT"
}
```

**Comment by rainboyan** [comment]
Fixed via 6295afe86fa6bb5a5c9ae5fb0eebb15ad5f43e53
